### PR TITLE
Add new frontend system support for datadog plugin

### PR DESCRIPTION
**Pull request by alex-mckay**
This PR adds support for the [new frontend system](https://backstage.io/docs/frontend-system/) to the datadog plugin by exposing entity cards, entity content, and the API as extensions under the '/alpha' subpath. This change is fully back-compatible / has no affect on consumers who are still using the standard frontend system.

Spotify will be updating OSS plugins to the new frontend & backend as we work towards an updated Backstage plugin framework. We're prioritizing some community favorites first, and the Spotify team will help manage any unintentional issues that arise.

#### :heavy_check_mark: Checklist

- [X] Added tests for new functionality and regression tests for bug fixes
- [X] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/frontend/backstage-plugin-datadog/src/components/GraphWidget.tsx [hunk 1]
```diff
@@ -15,8 +15,10 @@
  */
 
 import { Entity } from '@backstage/catalog-model';
-import { useEntity } from '@backstage/plugin-catalog-react';
-import { MissingAnnotationEmptyState } from '@backstage/core-components';
```
Comment (alex-mckay) [inline plugins/frontend/backstage-plugin-datadog/src/components/GraphWidget.tsx:19]: MissingAnnotationEmptyState from core-components is deprecated / shifted to importing the same thing from plugin-catalog-react instead

**Comment by alex-mckay** [comment]
Hey @Xantier - looking for a review whenever you have a chance! Can't tell if the 'RoadieHQ/engineering' actually has members or not so tagging just to be safe - but feel free to redirect me if you're not the right maintainer :)

**Comment by alex-mckay** [comment]
> Hey @Xantier - looking for a review whenever you have a chance! Can't tell if the 'RoadieHQ/engineering' actually has members or not so tagging just to be safe - but feel free to redirect me if you're not the right maintainer :)



> Hey @Xantier - looking for a review whenever you have a chance! Can't tell if the 'RoadieHQ/engineering' actually has members or not so tagging just to be safe - but feel free to redirect me if you're not the right maintainer :)

appreciate the CI trigger! fixed the issue where yarn.lock was pointing at the wrong registry for the new deps, should be ready for a re-run!

File: .changeset/wicked-months-listen.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@roadiehq/backstage-plugin-datadog': patch
```
Comment (Xantier) [inline .changeset/wicked-months-listen.md:2]: This looks to be introducing new functionality so we should be using minor version here.

File: plugins/frontend/backstage-plugin-datadog/package.json [hunk 3]
```diff
@@ -5,12 +5,29 @@
...
   "license": "Apache-2.0",
   "publishConfig": {
-    "access": "public",
-    "main": "dist/index.esm.js",
-    "types": "dist/index.d.ts"
+    "access": "public"
+  },
+  "exports": {
+    ".": "./src/index.ts",
+    "./alpha": "./src/alpha.ts",
```
Comment (Xantier) [inline plugins/frontend/backstage-plugin-datadog/package.json:12]: Is this the naming convention for this? We need to eventually modify this to be something else than `alpha` if it ever graduates to stable, or remove it completely if it doesn't. Do we have a maintenance plan to bump this (as a breaking change) into beta and stable stages?

Comment (alex-mckay) [reply plugins/frontend/backstage-plugin-datadog/package.json:12]: Yes, this is the naming convention for the system for now while it's in alpha stage (docs [here](https://backstage.io/docs/frontend-system/building-plugins/migrating#migrating-the-plugin)).Alluded to this in the PR description, but Spotify will own the process of defining the upgrade path/maintenance once the frontend system moves out of alpha.

**Comment by alex-mckay** [comment]
@Xantier addressed your question, and fixed the net-new dependency versions to match this repo's 1.26.5 version of backstage (apologies for missing before that, I had been running yarn test from within the plugin directory, where the incompatible versions weren't surfaced in the way they do during the [test CI run](https://github.com/RoadieHQ/roadie-backstage-plugins/actions/runs/10202503235/job/28251891233?pr=1520)). Let me know if you have additional questions about Spotify's role in supporting the [new frontend system](https://backstage.io/docs/frontend-system/) / I can always loop in some of our maintainers.

**Review by Xantier — APPROVED** [review]

**Pull request merged**